bson/src/main/org/bson/BsonDocument.java

```diff
@@ -771,6 +771,17 @@ public BsonDocument append(final String key, final BsonValue value) {
         return this;
     }
 
+    /**
+     * Gets the first key in the document.
+     *
+     * @return the first key in the document
+     * @throws java.util.NoSuchElementException if the document is empty
+     * @since 3.6
+     */
+    public String getFirstKey() {
```

Comment:
Needed so that InternalStreamConnection can just call it without having to know what subclass of BsonDocument it has.

Reply:
Ack